docs/elasticsearch-spec.asciidoc

```diff
@@ -0,0 +1,103 @@
+[id="{p}-elasticsearch-specification"]
+== Elasticsearch Specification
+
+There are a number of settings which need to be considered before going into production related to Elasticsearch but also to Kubernetes.
+
+Basic settings
+
+- JVM heap size
+- Node configuration
+- HTTP settings & TLS SANs
+- Resource limits
+- Pod Template
+- Volume claim templates
+
+Advanced settings
+
+- Virtual memory
+- Custom HTTP certificate
+- Secure settings
+- Custom plugins and bundles
+- Init containers for plugin downloads
+- Pod disruption budget
+- Change budget (maxUnavailable, maxSurge)
+
+[id="{p}-node-configuration"]
+=== Node configuration
+
+Any setting that can be configured in the `elasticsearch.yml` configuration file can be defined for each topology of nodes in the `spec.nodes[?].config` section.
+
+[source,yaml]
+----
+kind: Elasticsearch
```

Comment:
I think we should either include `Kind` + `apiVersion` + `metadata`, either none of those but only the `spec`. An in-between choice can be confusing.

Reply:
I forget to remove it.
This is because at the beginning I documented ES and Kibana.